spec/rubocop/cop/lint/return_in_void_context_spec.rb

```diff
@@ -12,6 +12,58 @@ def initialize
         end
       RUBY
     end
+
+    it 'registers an offense when the value is returned in a block' do
+      expect_offense(<<~RUBY)
+        class A
+          def initialize
+            foo do
+              return :qux
+              ^^^^^^ Do not return a value in `initialize`.
+            end
+          end
+        end
+      RUBY
+    end
+
+    it 'registers an offense when the value is returned in a numblock' do
+      expect_offense(<<~RUBY)
+        class A
+          def initialize
+            foo do
+              _1
+              return :qux
+              ^^^^^^ Do not return a value in `initialize`.
+            end
+          end
+        end
+      RUBY
+    end
+
+    it 'registers an offense when the value is returned from inside a proc' do
+      expect_offense(<<~RUBY)
+        class A
+          def initialize
+            Proc.new do
```

Comment:
Is the intended explanation referring to `proc`?

```suggestion
            proc do
```

Reply:
Either works, I don't think there is any difference between the two. I've updated to `proc` to be more inline with the spec description (it's also what `Style/Proc` would want).

Reply:
Yep, the behavior is the same in both cases. This is about ensuring the spec description matches the example.